protocols/delegate-manager/contracts/DelegateManager.sol

```diff
@@ -0,0 +1,118 @@
+/*
+  Copyright 2019 Swap Holdings Ltd.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+pragma solidity 0.5.10;
+pragma experimental ABIEncoderV2;
+
+import "@airswap/indexer/contracts/interfaces/IIndexer.sol";
+import "@airswap/swap/contracts/interfaces/ISwap.sol";
+import "@airswap/delegate-factory/contracts/interfaces/IDelegateFactory.sol";
+import "@airswap/delegate/contracts/interfaces/IDelegate.sol";
+import "@airswap/types/contracts/Types.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+
+contract DelegateManager is Ownable {
+
+    IDelegateFactory public factory;
+    IDelegate public delegate;
+
+    /**
+      * @notice Constructor to create the Delegate Manager
+      * @dev factory is unable to be changed after the DelegateManager is constructed
+      * @param _factory address of the factory contract that will create the delegates
+      * @param _tradeWallet the wallet that the delegate will be trading on behalf of
+      */
+    constructor(IDelegateFactory _factory, address _tradeWallet) public {
```

Comment:
Just a dump of some thoughts...
The reason we have a DelegateFactory is to make it easier for people to deploy themselves a new, trusted Delegate on-chain with just the call of a function. Obviously the DelegateFactory brings extra advantages in that you can tell on-chain whether a Delegate is a trusted delegate. However it seems to make it more complex for a user now that they have to deploy an entire contract that then calls the factory function in the constructor... If we stick with a DelegateManager being 1-1 with a Delegate, it could be that we want a DelegateManagerFactory too?

Reply:
Ooh or maybe 1 factory that does both? You could have 2 functions
1) one that just deploys a Delegate, as our current factory does
1) one that deploys a Delegate and a Manager to go with it (the order of the deploys and how the contracts become linked would need to be thought about, but it could be neat fi we get it right)

Reply:
I'm open to the above ideas.